src/raftstore/store/snap.rs

```diff
@@ -618,6 +620,987 @@ mod v1 {
+    }
+
+
+    fn decode_cf_size_checksums(kvs: &[KeyValue]) -> RaftStoreResult<Vec<(CfName, u64, u32)>> {
+        let snapshot_cfs = get_snapshot_cfs();
+        let mut cf_sizes: Vec<(CfName, u64)> = vec![];
+        let mut cf_checksums: Vec<(CfName, u32)> = vec![];
+        for kv in kvs {
+            let mut key_bytes = kv.get_key();
+            let decoded = try!(key_bytes.decode_bytes(ENCODE_DECODE_DESC));
+            let key = match String::from_utf8(decoded.clone()) {
+                Ok(s) => s,
+                Err(e) => {
+                    return Err(box_err!("fail to parse snapshot meta key {:?}, err: {:?}",
+                                        decoded,
+                                        e));
+                }
+            };
+            let strs: Vec<&str> = key.split('_').collect();
+            if strs.len() != 2 {
+                return Err(box_err!("invalid snapshot meta key {}", key));
+            }
+            let cf_name = strs[1];
+            let res = snapshot_cfs.iter().find(|&s| *s == cf_name);
+            if res.is_none() {
+                return Err(box_err!("invalid snapshot cf {}", cf_name));
+            }
+            let cf = res.unwrap();
+            match strs[0] {
+                SNAPSHOT_META_PREFIX_SIZE => {
+                    let size = try!(kv.get_value().decode_u64());
+                    cf_sizes.push((cf, size));
+                }
+                SNAPSHOT_META_PREFIX_CHECKSUM => {
+                    let value = try!(kv.get_value().decode_u64());
+                    if value > u32::max_value() as u64 {
+                        return Err(box_err!("invalid snapshot checksum {} for cf {}", value, cf));
+                    }
+                    cf_checksums.push((cf, value as u32));
+                }
+                _ => return Err(box_err!("invalid snapshot meta prefix {}", strs[0])),
+            }
+        }
+        if cf_sizes.len() != snapshot_cfs.len() {
+            return Err(box_err!("invalid number of snapshot meta cf size: {}, expected: {}",
+                                cf_sizes.len(),
+                                snapshot_cfs.len()));
+        }
+        if cf_checksums.len() != snapshot_cfs.len() {
+            return Err(box_err!("invalid number of snapshot meta cf checksum: {}, expected: {}",
+                                cf_checksums.len(),
+                                snapshot_cfs.len()));
+        }
+        let mut cf_size_checksums = Vec::with_capacity(snapshot_cfs.len());
+        for (cf, size) in cf_sizes {
+            let checksum = cf_checksums.iter().find(|c| c.0 == cf).unwrap().1;
+            cf_size_checksums.push((cf, size, checksum));
+        }
+        Ok(cf_size_checksums)
+    }
+
+    #[derive(Default)]
+    struct CfFile {
+        pub cf: CfName,
+        pub path: String,
+        pub size: u64,
+
+        // for building snapshot
+        pub sst_writer: Option<SstFileWriter>,
+        pub kv_count: u64,
```

Comment:
do we add snapshot kv count and size to the SNAPSHOT_KV_COUNT_HISTOGRAM and SNAPSHOT_SIZE_HISTOGRAM?